src/gzip/header.cr

```diff
@@ -85,8 +85,8 @@ class Gzip::Header
     # os
     io.write_byte os
 
-    if extra = @extra
-      io.write_byte extra.size.to_u8
+    unless @extra.empty?
+      io.write_byte @extra.size.to_u8
       io.write(extra)
```

Comment:
Think you're right. It used to copy the value, I'm guessing their isn't a test case for gzip with extra set. I'll update.